### ISSUE TITLE
Feature request chkdsk

### ISSUE THREAD

**Issue by mkools** [FSLogix/Invoke-FslShrinkDisk#25]
Hi,

I've noticed that sometimes a profile disk is e.g. 15 GB in size. Windows reports only 2.5 GB is used and for some reason the shrink script can't do anything with it. I did a manual mount of the VHDX and tried to shrink it with windows but then it reported and error and said I should run chkdsk on the VHDX. I did and it found and fixed some errors, after that I was able to shrink it to a 4 GB file.

So long story short, my question is if there could be a checkdisk option added to the script, which mounts it, runs chkdsk /f and only then does the resizing?

**Comment by JimMoyle** [comment]
This feature is a good idea and is already implemented in the development
branch.

**Comment by chrismoore-28** [comment]
Hi Jim, 

did the chkdsk feature get added in? I checked over the Dev branch but nothing was present for it.

**Comment by Tiger-Nuts** [comment]
I would love to have this feature too. Is there a reason why it didn't get in?

**Comment by zimbo86** [comment]
same issue here.
Many large profiles with whitespace. Noticed today, that they can´t be shrinked, because GetSupportedSize wont run.
After a chkdsk repair everything works and the disk is successfully shrinked.

**Comment by AlexCloudiction** [comment]
Hi,
We still experience issues with the chkdsk issue. Is there a solution for this? we have a lot of corrupted disk.
